src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -363,6 +365,67 @@ library GPv2Encoding {
     /// ```
     ///
     /// Unused signature data is returned along with the address of the signer.
+    /// If the encoding is not valid, for example because the calldata does not
+    /// suffice, the function reverts.
+    ///
+    /// @param encodedSignature Calldata pointing to tightly packed signature
+    /// bytes.
+    /// @return r r parameter of the ECDSA signature.
+    /// @return s s parameter of the ECDSA signature.
+    /// @return v v parameter of the ECDSA signature.
+    /// @return remainingCalldata Input calldata that has not been used to
+    /// decode the signature.
+    function decodeEcdsaSignature(bytes calldata encodedSignature)
```

Comment:
nanonit: a separate refactoring could move all the signing schemes in a GPv2Signing.sol library given that this one is becoming quite large.